modules/core/src/main/java/io/fluo/recipes/map/CollisionFreeMap.java

```diff
@@ -168,12 +185,16 @@ public U apply(Bytes b) {
         } else {
           tx.set(currentValueRow, DATA_COLUMN, newVal);
         }
-        V cv = null;
-        if (currVal != null) {
-          // deserialize again in case combiner mutated Object
-          cv = serializer.deserialize(currVal.toArray(), valType);
+
+        if (partiallyProcessedKey == null || !partiallyProcessedKey.equals(entry.getKey())) {
+          V cv = null;
+          if (currVal != null) {
+            // deserialize again in case combiner mutated Object
+            cv = serializer.deserialize(currVal.toArray(), valType);
+          }
+
+          updatesToReport.add(new Update<K, V>(kd, cv, nv.orNull()));
```

Comment:
In #31 I modified the CFM to combine and push back to the update queue instead of updating the current value